modelskill/comparison/_comparison.py

```diff
@@ -369,6 +369,12 @@ def _matched_data_to_xarray(
     cols = list(df.columns)
     items = ItemSelection.parse(cols, obs_item, mod_items, aux_items, x_item, y_item)
 
```

Comment:
Maybe add extra check that `x` (and similar for `y`) that they are floating point numbers. It should be obvious from the type hint and the docstring, but still...
```suggestion
if x is not None:
  try:
    x = float(x)
  except:
    raise ValueError(f"x must be a scalar number, not {type(x)}"
```

Reply:
Good idea - I was just working on a failing test - I will implement this. 

And I guess we do not want and "except all" so what is the right error TypeError? ValueError?

Reply:
if the type is wrong, I guess `TypeError` is what we want and not `ValueError`😐